activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -476,6 +476,8 @@ def get_period_and_ensure_valid_local_time(period)
       end
 
       def transfer_time_values_to_utc_constructor(time)
+        # avoid creating another Time object if possible
+        return time if time.utc? && time.is_a?(Time)
```

Comment:
You're welcome! I realized one last thing just now.

The `instance_of?` call needs to have the `Time` class rooted by adding `::` in front of the class name (just like they did a few lines down).

Since this method is within `ActiveSupport::TimeWithZone`, `time.instance_of?(Time)` would return true if `time` were a `ActiveSupport::Time`, `ActiveSupport::TimeWithZone::Time`, or a plain old `Time`. To prevent against this and ensure that only `Time` objects return true, the line should actually be:

```ruby
return time if time.instance_of?(::Time) && time.utc?
```